### ISSUE TITLE
Cannot read property 'next' of undefined when stepping over destructuring assignment

### ISSUE THREAD

**Issue by miluoshi** [Microsoft/vscode-chrome-debug#435]
- VS Code Version: 1.12.2
- Debugger for Chrome: 3.1.1
- NodeJS Version: 6.5.0

Steps to reproduce:

1. Set a breakpoint on a line with a destructuring array assignment
2. Run the app and wait to reach the breakpoint
3. Step over the breakpoint
4. The app crashes

The node app crashes with a following error on the line with the destructuring assignment: `TypeError: Cannot read property 'next' of undefined at Q.all.then.response`

Example code:
```javascript
let updatedTicketPromise = Q.all([userOrgsPromise, ticketPromise])
  .then(response => {
    const [userOrgs, ticket] = response // The error is on this line
    ...
```

I am not sure whether this is an error of vscode-chrome-debug or Q module, but it happens ONLY on a line with destructuring assignment and only if breakpoint is set and stepped over on this line.

**Comment by roblourens** [comment]
That error would be coming from the app, possibly a node bug if it only happens while debugging. Is that the actual script that executes, or is there a build process, like with babel?

**Comment by miluoshi** [comment]
It's the actual script.

**Comment by roblourens** [comment]
Ok, well I'm not sure where `next` is coming from given that code but I'd check whether you get the same thing when using chrome devtools.

**Comment by eliprand** [comment]
FWIW,
We ran into this issue today. It wasn't me but one of my team member. Similar to @miluoshi 's case, the line of code that was throwing the `TypeError: Cannot read property 'next' of undefined` did not have any reference to `next()` and had been working fine for months.
Eventually, I noticed that my team member was using VSCode launch capabilities which was starting NodeJS (`v6.11.0`) with the `--nolazy` option.
I had them switch to manually starting (with `--debug` or `--inspect`) and then attach and the problem went away.
I am freaking out a little bit as I can't tell if the error is because we are debugging or because of the `--nolazy` option. I will do some additional digging if possible, but any feedback would be greatly appreciated here.
